plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/types/EdgeTypeHint.java

```diff
@@ -25,12 +27,19 @@ public class EdgeTypeHint extends ElementTypeHint {
    private boolean routable;
    private List<String> sourceElementTypeIds;
    private List<String> targetElementTypeIds;
+   private boolean dynamic;
 
    public EdgeTypeHint() {}
 
    public EdgeTypeHint(final String elementTypeId, final boolean repositionable, final boolean deletable,
       final boolean routable,
       final List<String> sourceElementTypeIds, final List<String> targetElementTypeIds) {
+      this(elementTypeId, repositionable, deletable, routable, false, sourceElementTypeIds, targetElementTypeIds);
+   }
+
+   public EdgeTypeHint(final String elementTypeId, final boolean repositionable, final boolean deletable,
```

Comment:
Not directly related to you change, but while we are at it we should probably make sure that the `createDefaultEdgeTypeHint` method of the `DiagramConfiguration` passes new, empty Lists for the source and target ids.
https://github.com/eclipse-glsp/glsp-server/blob/a48db7534ae1a729a3141ae50080b9c05ceb7eb5/plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/diagram/DiagramConfiguration.java#L74

Otherwise the corresponding properties in the `SetTypeHintsAction` are not defined which can cause issues on the client